x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/expandable_section/expandable_section_results.tsx

```diff
@@ -227,13 +232,56 @@ export const ExpandableSectionResults: FC<ExpandableSectionResultsProps> = ({
   );
 
   const openCustomUrl = (item: DataGridItem, customUrl: MlKibanaUrlConfig) => {
```

Comment:
This function has quite some overlap now with the same one in `x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx`, might be worth consolidating the shared logic? If we break out the inner shared logic into another utility it might also be easier to write jest unit tests for it, for example just for the time range stuff.

Reply:
This has been simplified and updated to no longer use 'interval' as in AD - not sure if it's worth trying to combine anymore.